tasks/main.yml

```diff
@@ -69,8 +69,22 @@
       ansible_distribution != 'Fedora' and
       ansible_distribution_major_version | int > 8
 
+- name: Ask user's consent to restart NetworkManager due to wireless or team
+    interfaces
+  fail:
+    msg: The user needs to consent to restart NetworKManager explicitly
+      by setting `network_allow_restart` to `true` when network packages
+      are changed and wireless or team connections are specified
```

Comment:
Please change this to imperative:

"NetworkManager needs to be restarted to be able to proceed. This might disturb the connectivity of the managed system. Please set `network_allow_restart` to `true` if you are prepared for this."

Writing this makes me realize that this probably won't work because when the user set the variable to true, then the role won't need to install the packages anymore, therefore it won't restart NM and it will fail afterwards. So it seems that NM needs to be probed to check if it needs to be restarted regardless of whether the packages were just installed.

Reply:
This also highlights why it is important to have a test case that covers the scenario.

Reply:
IMHO, this should be done before installing the packages if possible. At that point we can detect if `network_allow_restart` is set to `true`. Then, the user gets the error and can decide to do it or not.

Reply:
A tentative approach could be using `update_cache`, but then we need to write a task  for different package managers (`yum`, `dnf`), like:

```
    - name: Check if updates are available for network packages 
      dnf:
        update_cache: yes
        name: "{{ network_packages }}"
        state: latest
      register: package_info
      check_mode: yes 
      when: 
         -  ansible_distribution == 'Fedora' or ansible_distribution_major_version | int > 7  
      
    - name: Check if updates are available for network packages 
      yum:
        update_cache: yes
        name: "{{ network_packages }}"
        state: latest
      register: package_info
      check_mode: yes 
      when: 
         -   ansible_distribution != 'Fedora' and  ansible_distribution_major_version | int < 8      
```